src/app/api/services/add-to-bundle/add-to-bundle.service.ts

```diff
@@ -0,0 +1,67 @@
+import { CoreWebService, ApiRoot, ResponseView } from 'dotcms-js/dotcms-js';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { RequestMethod } from '@angular/http';
+import { DotCurrentUser } from '../../../shared/models/dot-current-user/dot-current-user';
+import { DotBundle } from '../../../shared/models/dot-bundle/dot-bundle';
+import { AjaxActionResponseView } from '../../../shared/models/ajax-action-response/ajax-action-response';
+
+@Injectable()
+export class AddToBundleService {
+    private currentUsersUrl = 'v1/users/current/';
+    private bundleUrl = `bundle/getunsendbundles/userid`;
+    /*
+        TODO: I had to do this because this line concat 'api/' into the URL
+        https://github.com/dotCMS/dotcms-js/blob/master/src/core/core-web.service.ts#L169
+    */
+    private addToBundleUrl = `${this._apiRoot.baseUrl}DotAjaxDirector/com.dotcms.publisher.ajax.RemotePublishAjaxAction/cmd/addToBundle`;
+
+    constructor(public _apiRoot: ApiRoot, private coreWebService: CoreWebService) { }
+
+    /**
+     * Get bundle items
+     * @returns {Observable<any[]>}
+     * @memberof AddToBundleService
+     */
+    getBundle(): Observable<any[]> {
+        return this.getCurrentUser().mergeMap((user: DotCurrentUser) => {
+            return this.coreWebService.requestView({
+                method: RequestMethod.Get,
+                url: `${this.bundleUrl}/${user.userId}`
+            }).map((res: any) => JSON.parse(res.resp._body));
+        })
+        .flatMap((bundle: any) => bundle.items)
```

Comment:
Why are you doing a flatmap here? 

Why don't you use DotBundle here?

Reply:
I just refactor the code, removing flatMap adding ```.pluck('bodyJsonObject', 'items');```